check for `go generate` on CircleCI

tbg opened the pull request:
this guarantees that no commit for which `go generate ./...` modifies or
deletes any files passes the test.

tbg committed:
make up for forgotten `go generate`

tbg committed:
sort files in proto `make all`

this helps to ensure that `go generate` becomes
reproducible.

petermattis commented:
LGTM

bdarnell commented on circle.yml:
```diff
@@ -30,6 +30,8 @@ dependencies:
 test:
   override:
+    # Check whether the committer forgot to run `go generate`.
+    - docker run cockroachdb/cockroach-dev shell "go generate ./... && git ls-files --modified --deleted | diff /dev/null -"
```
I think the git ls-files call should also include --others to find new/untracked files.

tschottdorf replied:
That'll require some more changes since we add cockroach.sh in the directory, but I'll just move that to the root before I merge this.

bdarnell replied:
Or you can add cockroach.sh to .gitignore (with a comment about when this occurs).

tschottdorf replied:
done.

bdarnell commented:
LGTM (I also see that the github change to treat .pb.go files as generated has gone live)

tbg committed:
check for `go generate` on CircleCI

this guarantees that no commit for which `go generate ./...` modifies or
deletes any files passes the test.
